chia/consensus/multiprocess_validation.py

```diff
@@ -209,14 +210,19 @@ async def pre_validate_blocks_multiprocessing(
             if curr.first_in_sub_slot:
                 assert curr.finished_challenge_slot_hashes is not None
                 num_sub_slots_found += len(curr.finished_challenge_slot_hashes)
-            recent_blocks[curr.header_hash] = curr
+            recent_blocks[header_hash] = curr
             if curr.is_transaction_block:
                 num_blocks_seen += 1
-            curr = block_records.block_record(curr.prev_hash)
-        recent_blocks[curr.header_hash] = curr
+            header_hash = curr.prev_hash
```

Comment:
once we follow the link to the previous block, we know the next header hash will be the same as `prev_hash`, so just save that instead of re-computing it every time